lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -955,138 +986,166 @@ def determine_agent_level_common_labels
+      case resource.type
       # Cloud Functions.
       when CLOUDFUNCTIONS_CONSTANTS[:resource_type]
-        group_resource_labels.merge!(
+        resource.labels.merge!(
           'region' => @gcf_region,
           'function_name' => decode_cloudfunctions_function_name(
-            matched_regex_group['encoded_function_name'])
+            matched_regexp_group['encoded_function_name'])
         )
-
-        instance_id = group_resource_labels.delete('instance_id')
-        group_common_labels.merge!(
-          "#{CONTAINER_CONSTANTS[:service]}/instance_id" => instance_id,
+        instance_id = resource.labels.delete('instance_id')
+        common_labels.merge!(
+          "#{GKE_CONSTANTS[:service]}/instance_id" => instance_id,
           "#{COMPUTE_CONSTANTS[:service]}/resource_id" => instance_id,
-          "#{CONTAINER_CONSTANTS[:service]}/cluster_name" =>
-            group_resource_labels.delete('cluster_name'),
+          "#{GKE_CONSTANTS[:service]}/cluster_name" =>
+            resource.labels.delete('cluster_name'),
           "#{COMPUTE_CONSTANTS[:service]}/zone" =>
-            group_resource_labels.delete('zone')
+            resource.labels.delete('zone')
         )
 
       # GKE container.
-      when CONTAINER_CONSTANTS[:resource_type]
-        if matched_regex_group
+      when GKE_CONSTANTS[:resource_type]
+        if matched_regexp_group
           # We only expect one occurrence of each key in the match group.
           resource_labels_candidates =
-            matched_regex_group.names.zip(matched_regex_group.captures).to_h
-          common_labels_candidates =
-            resource_labels_candidates.dup
-          group_resource_labels.merge!(
+            matched_regexp_group.names.zip(matched_regexp_group.captures).to_h
+          common_labels_candidates = resource_labels_candidates.dup
+          resource.labels.merge!(
             delete_and_extract_labels(
               resource_labels_candidates,
               # The kubernetes_tag_regexp is poorly named. 'namespace_name' is
               # in fact 'namespace_id'. 'pod_name' is in fact 'pod_id'.
               # TODO(qingling128): Figure out how to put this map into
-              # constants like CONTAINER_CONSTANTS[:extra_resource_labels].
+              # constants like GKE_CONSTANTS[:extra_resource_labels].
               'container_name' => 'container_name',
               'namespace_name' => 'namespace_id',
               'pod_name' => 'pod_id'))
 
-          group_common_labels.merge!(
+          common_labels.merge!(
             delete_and_extract_labels(
               common_labels_candidates,
-              CONTAINER_CONSTANTS[:extra_common_labels]
-                .map { |l| [l, "#{CONTAINER_CONSTANTS[:service]}/#{l}"] }.to_h))
+              GKE_CONSTANTS[:extra_common_labels]
+                .map { |l| [l, "#{GKE_CONSTANTS[:service]}/#{l}"] }.to_h))
         end
+
+      # Docker container.
+      when DOCKER_CONSTANTS[:resource_type]
```

Comment:
Per discussion with Summit, we should not be sending this metadata label for docker container since it's been ingested in the Logging Backend. For other existing resource types we have been doing it for a while and could not easily deprecate it.

Reply:
> *Aside*: I feel we're introducing more legacy stuff to get rid of by propagating `DOCKER_CONSTANTS` throughout the code. Ideally, we shouldn't need to reference it at all -- the code should not care what the resource looks like.

That said, in this particular case we shouldn't need to remove the `resource_name` field because it would never be added (lines 975-976 should only be executed on raw GCE or on GKE). So this seems like dead code to me, unless I'm missing something.

Reply:
The line 975-976 gets executed on agent startup. At that point we can't tell `a raw GCE instance` from `a GCE instance with docker installed and potentially want to send docker logs`. The we will be adding this `resource_name` label regardless.

`Docker container` logs only get identified when the monitored resource (specified by `logging.googleapis.com/local_resource_id`) returned by metadata agent has `resource_type` as `docker_container`. This process happens kind of late when logs start coming in.

In a word, we still need to remove `resource_name` for `docker_container` unless we change the behavior dramatically as proposed in b/65175256.

Reply:
Fine, makes sense. Can we add a TODO to remove this once the resource is retrieved at a proper time?

Reply:
Added.